core/src/main/scala/kafka/message/ByteBufferMessageSet.scala

```diff
@@ -492,7 +497,9 @@ class ByteBufferMessageSet(val buffer: ByteBuffer) extends MessageSet with Loggi
           Utils.writeUnsignedInt(buffer, MessageSet.LogOverhead + Message.CrcOffset, wrapperMessage.computeChecksum)
         }
         buffer.rewind()
-        (this, false)
+        // For compressed messages,
```

Comment:
For the code btw 484 and 491, if messageTimestampType and timestamp != maxTimestamp, don't we need to change the timestamp in the wrapper message?